poly-commitment/src/commitment.rs

```diff
@@ -1089,20 +1091,20 @@ pub mod caml {
 
     impl<G, CamlG> From<PolyComm<G>> for CamlPolyComm<CamlG>
     where
-        G: AffineCurve,
+        G: AffineRepr,
         CamlG: From<G>,
     {
         fn from(polycomm: PolyComm<G>) -> Self {
             Self {
-                unshifted: polycomm.unshifted.into_iter().map(Into::into).collect(),
-                shifted: polycomm.shifted.map(Into::into),
+                unshifted: polycomm.unshifted.into_iter().map(From::from).collect(),
+                shifted: polycomm.shifted.map(From::from),
```

Comment:
this change is intriguing

Reply:
Yeah, Rust compiler is... demanding. These are just black magic spells which seem to please The 🦀.

Reply:
I don't fully understand what's going on here. According to the docs, if `From` is implemented, Rust is supposed to give you `Into` automatically. For some reason `Into` didn't work here, so I tried `From` and it worked, so here it is 🙃

Reply:
Here's the exact error message if these `From::from` are changed back to `Into::into`:

```
error[E0277]: a value of type `Vec<CamlG>` cannot be built from an iterator over elements of type `<G as ark_ec::AffineRepr>::Group`
    --> poly-commitment/src/commitment.rs:1094:28
     |
1094 |                 unshifted: polycomm.unshifted.into_iter().map(Into::into).collect(),
     |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ------- required by a bound introduced by this call
     |                            |
     |                            value of type `Vec<CamlG>` cannot be built from `std::iter::Iterator<Item=<G as ark_ec::AffineRepr>::Group>`
     |
     = help: the trait `FromIterator<<G as ark_ec::AffineRepr>::Group>` is not implemented for `Vec<CamlG>`
note: required by a bound in `std::iter::Iterator::collect`
    --> /Users/ivan/.rustup/toolchains/1.67.0-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:1832:19
     |
1832 |     fn collect<B: FromIterator<Self::Item>>(self) -> B
     |                   ^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `std::iter::Iterator::collect`
help: consider extending the `where` clause, but there might be an alternative better way to express this requirement
     |
1090 |         CamlG: From<G>, Vec<CamlG>: FromIterator<<G as ark_ec::AffineRepr>::Group>
     |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```